app/src/main/java/com/mdg/droiders/samagra/shush/AlarmScheduler.java

```diff
@@ -30,52 +36,299 @@ public AlarmScheduler(Context context) {
+     *              For e.g. - Monday is 0, Tuesday is 1, ... and so on
+     * @param rowID Unique primary key of the shush alarm row.
+     */
+    public void cancelSingleAlarm(int day, Integer rowID) {
+        if (rowID == null) {
+            return;
+        }
+        alarmManager.cancel(getDefaultPendingIntent(getStartDayID(day, rowID)));
+        alarmManager.cancel(getDefaultPendingIntent(getEndDayID(day, rowID)));
+    }
+
+    /**
+     * Sets a weekly alarm to silence your phone for days given by boolean array days.
+     * <br>
+     * <b>NOTE</b> : This method only schedules alarms and does NOT cancels for some day whose
+     * corresponding index in days array is false. To cancel alarms
+     * use {@link #cancelSingleAlarm(int, Integer)}
+     * <br><br>
+     * <b>Uses</b> : {@link #setAlarm(long, long, Integer, Integer)} to set each alarm one by one.
+     *
+     * @param startTime The time of day at which phone is to be shushed
+     * @param endTime   The time of day at which phone is to be un-shushed
+     * @param days      A boolean array that is true if the alarm is to be
+     *                  scheduled for the corresponding day. i.e. days[0] is set to be true
+     *                  then a weekly alarm will be set such that it will shush the phone every Monday.
+     * @param rowID     Unique primary key of the shush alarm row.
+     */
+    public void setWeeklyAlarm(Calendar startTime, Calendar endTime, boolean[] days, Integer rowID) {
+        // To avoid various exceptions
+        if (rowID == null || days.length != 7) {
+            return;
+        }
+        // Instance of calendar that holds current system time
+        Calendar rightNow = Calendar.getInstance();
+
+        // Set YEAR, MONTH and DAY fields of startTime
+        // and endTime to today's YEAR, MONTH and DAY respectively
+        startTime.set(Calendar.YEAR, rightNow.get(Calendar.YEAR));
+        startTime.set(Calendar.MONTH, rightNow.get(Calendar.MONTH));
+        startTime.set(Calendar.DAY_OF_MONTH, rightNow.get(Calendar.DAY_OF_MONTH));
+        endTime.set(Calendar.YEAR, rightNow.get(Calendar.YEAR));
+        endTime.set(Calendar.MONTH, rightNow.get(Calendar.MONTH));
+        endTime.set(Calendar.DAY_OF_MONTH, rightNow.get(Calendar.DAY_OF_MONTH));
+        Log.d(LOG_TAG + "now", LOG_DATE_FORMAT.format(rightNow.getTime()));
+
+        // If startTime's hour is ahead of endTime's, assume that
+        // endTime is of that of next day
+        if (startTime.get(Calendar.HOUR_OF_DAY)
+                > endTime.get(Calendar.HOUR_OF_DAY)) {
+            endTime.add(Calendar.DAY_OF_MONTH, 1);
+        }
+
+        // If endTime is in the past increment both start and end time
+        if (endTime.compareTo(rightNow) < 0) {
+            startTime.add(Calendar.DAY_OF_MONTH, 1);
+            endTime.add(Calendar.DAY_OF_MONTH, 1);
+        }
+
+        // get Day index of start Time
+        int i = getDay(startTime);
```

Comment:
Though nitpicking using i for variables that are used method wide isn't a good choice IMO

Reply:
How can this cause a problem? Can you please elaborate?